spec.html

```diff
@@ -11059,6 +11060,44 @@ <h1>Examples of Automatic Semicolon Insertion</h1>
+
+    <p>The term &ldquo;<dfn>automatic semicolon insertion hazard</dfn>&rdquo; refers, to a place where a semicolon may or may not be inserted depending on preceding source text according to the rules above. These places are considered hazards if changes outside of an existing source text grammar production could change the grammar production of the existing source text depending on automatic semicolon insertion. The rest of this section describes a number of automatic semicolon insertion hazards in this version of ECMAScript.</p>
+    <emu-clause id="sec-asi-hazards-in-statement-lists">
+      <h1>Automatic Semicolon Insertion Hazards in Statement Lists</h1>
+      <p>In a |StatementList|, many |StatementListItem|s end in semicolons, which may be omitted using automatic semicolon insertion. As a consequence of the rules above, at the end of a line ending an expression, a semicolon is required if the following line begins with any of the following:
+      <ul>
+        <li><strong>An opening parenthesis (<code>(</code>)</strong>. Without a semicolon, the two lines together are treated as a |CallExpression|.</li>
+        <li><strong>An opening square bracket (<code>[</code>)</strong>. Without a semicolon, the two lines together are treated as property access, rather than an |ArrayLiteral| or |ArrayAssignmentPattern|.</li>
+        <li><strong>A template string (<code>`</code>)</strong>. Without a semicolon, the two lines together are interpreted as a tagged Template (<emu-xref href="#sec-tagged-templates"></emu-xref>), with the previous expression as the |MemberExpression|.</li>
+        <li><strong>Unary <code>+</code> or <code>-</code></strong>. Without a semicolon, the two lines together are interpreted as a usage of the corresponding binary operator.</li>
+        <li><strong>A RegExp literal</strong>. Without a semicolon, the two lines together may be parsed instead as the `/` |MultiplicativeOperator|, for example if the RegExp has flags.</li>
+      </ul>
+    </emu-clause>
+  </emu-clause>
+  <emu-clause id="sec-no-lineterminator-here-automatic-semicolon-insertion">
+    <h1>Automatic Semicolon Insertion and &ldquo;[no |LineTerminator| here]&rdquo;</h1>
+    <em>This section is non-normative.</em>
+    <p>ECMAScript contains grammar productions which include &ldquo;[no |LineTerminator| here]&rdquo;. These productions are sometimes a means to have optional operands in the grammar. Introducing a |LineTerminator| in these locations would change the grammar production of a source text by using the grammar production without the optional operand. These locations are not considered hazardous because additive changes would need to alter source text within a grammar production that is subject to automatic semicolon insertion in order to change the grammar production.</p>
```

Comment:
I'm not sure if it's necessary to call some of these cases "hazardous" and others "not hazardous". Maybe it's best if we just refer directly to how these relate to code composition and authoring, and make up new terms for each of the two.

Reply:
I'm fine with renaming, but one causes errors from external productions and one requires productions to be edited to change behavior (something to be expected). Hazard / warning / etc. seems good if it could cause errors. Term can be changed as long as the difference in risk is preserved.